### ISSUE TITLE
Log new versions as they are found

### ISSUE THREAD

**Issue by rethab** [raineorshine/npm-check-updates#947]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 10.17`

---------------------------

It would be nice if new versions of packages would be printed out as they are found. I have tried `silly` mode, but even then I can only see new versions once the entire process was successful.

Background: For better or worse, I'm am using a not only slow but also unstable registry. This means `ncu -u` often times out (even when increasing the timeout by 10x) or fails for some other reason late in the process. This means I don't have any new versions at all even though ncu made 90% progress.

By printing new versions as they are discovered, I could at least fix some of the dependencies manually :)

**Comment by raineorshine** [comment]
I agree! That would be a better user experience. It involves some structural changes to the package fetching pipeline. I think it would be a good idea to tackle this after the conversion to typescript (#888).

**Comment by raineorshine** [comment]
There is the annoying point that the cli table won't work if they are output individually. I think the cli will need to re-render the whole table each time a new version is retrieved.

**Comment by lyswhut** [comment]
My network connection official npm registry is unstable,
A large number of requests sent by `ncu -u` occasionally have some connection failures, causing ncu to stop working, so adding connection failure retries can help alleviate this problem.
same: <https://github.com/raineorshine/npm-check-updates/issues/959>

![image](https://user-images.githubusercontent.com/20365169/141249889-fac0ce03-b295-473e-b84b-e4599fcca194.png)
